spec/models/vim_performance_state_spec.rb

```diff
@@ -73,4 +116,28 @@
       end
     end
   end
+
+  private
+
+  def create_past_present_future(factory, params, return_old: false)
+    # not deleted is captured
+    ret = FactoryBot.create_list(factory, 1, params)
+    # deleted before is not captured
+    old = FactoryBot.create_list(factory, 1, params.merge(:deleted_on => (capture_time - 2.hours)))
+    # deleted later is captured
+    ret << FactoryBot.create(factory, params.merge(:deleted_on => (capture_time + 2.hours)))
+
+    if return_old
+      [ret, old]
+    else
+      ret
+    end
+  end
+
+  # these represent records that have been pruned from the database
+  # so a find(id) will return nothing
+  def deleted_record_ids(objs)
+    base_id = objs.last.id
+    objs.size.times.map { |i| base_id + i + 1}
```

Comment:
Not that it matters much, but I think the following is a little simpler:

```suggestion
    base_id = objs.last.id
    (base_id + 1..base_id + objs.size).to_a
```

or, perhaps

```suggestion
    base_id = objs.last.id
    (base_id + 1).upto(base_id + objs.size).to_a
```

Reply:
thanks. I revisited and tried to make it even simpler `[objs.last.id + 1]`